core/src/main/java/org/apache/iceberg/DefaultValueParser.java

```diff
@@ -0,0 +1,190 @@
+        byte[] binaryBytes = BaseEncoding.base16().decode(defaultValue.textValue().toUpperCase(Locale.ROOT));
+        return ByteBuffer.wrap(binaryBytes);
+      case LIST:
+        Preconditions.checkArgument(defaultValue.isArray(),
+            "Cannot parse %s to a %s value", defaultValue, type);
+        List<Object> defaultList = Lists.newArrayList();
+        for (JsonNode element : defaultValue) {
+          defaultList.add(parseDefaultFromJson(type.asListType().elementType(), element));
+        }
+        return defaultList;
+      case MAP:
+        Preconditions.checkArgument(
+            defaultValue.isObject() && defaultValue.has("keys") && defaultValue.has("values") &&
+                defaultValue.get("keys").isArray() && defaultValue.get("values").isArray(),
+            "Cannot parse %s to a %s value",
+            defaultValue,
+            type);
+        Map<Object, Object> defaultMap = Maps.newHashMap();
+        JsonNode keys = defaultValue.get("keys");
+        JsonNode values = defaultValue.get("values");
+        List<JsonNode> keyList = Lists.newArrayList(keys.iterator());
+        List<JsonNode> valueList = Lists.newArrayList(values.iterator());
+
+        for (int i = 0; i < keyList.size(); i++) {
+          defaultMap.put(
+              parseDefaultFromJson(type.asMapType().keyType(), keyList.get(i)),
+              parseDefaultFromJson(type.asMapType().valueType(), valueList.get(i)));
+        }
+        return defaultMap;
+      case STRUCT:
+        Preconditions.checkArgument(defaultValue.isObject(),
+            "Cannot parse %s to a %s value", defaultValue, type);
+        Map<Integer, Object> defaultStruct = Maps.newHashMap();
+        for (Types.NestedField subField : type.asStructType().fields()) {
+          String fieldIdAsString = String.valueOf(subField.fieldId());
+          Object value = defaultValue.has(fieldIdAsString) ? parseDefaultFromJson(
+              subField.type(),
+              defaultValue.get(fieldIdAsString)) : null;
+          if (value != null) {
+            defaultStruct.put(subField.fieldId(), value);
+          }
+        }
+        return defaultStruct;
+      default:
+        return null;
+    }
+  }
+
+  public static Object convertJavaDefaultForSerialization(Type type, Object value) {
+    switch (type.typeId()) {
+      case DATE:
+        return DateTimeUtil.formatEpochDays((int) value);
+      case TIME:
+        return DateTimeUtil.formatTimeOfDayMicros((long) value);
+      case TIMESTAMP:
+        return DateTimeUtil.formatEpochTimeMicros((long) value, ((Types.TimestampType) type).shouldAdjustToUTC());
+      case FIXED:
+      case BINARY:
+        return BaseEncoding.base16().encode(((ByteBuffer) value).array());
+      case LIST:
+        List<Object> defaultList = (List<Object>) value;
+        List<Object> convertedList = Lists.newArrayListWithExpectedSize(defaultList.size());
+        for (Object element : defaultList) {
```

Comment:
This should call generator methods instead of returning `List<Object>`. Same with map and struct.

Reply:
I refactored the code and created a new method :

Reply:
No, as I said this should match the other parser classes and use a `JsonGenerator`.

Reply:
I refactored to use `JsonGenerator`.